flocker/node/agents/functional/test_cinder_behaviour.py

```diff
@@ -123,3 +110,104 @@ def test_updated_metadata_is_listed(self):
         missing_items = expected_items - actual_items
 
         self.assertEqual(set(), missing_items)
+
+
+def delete_multiple_volumes(cinder_volume_manager, volumes):
+    for volume in volumes:
+        cinder_volume_manager.delete(volume)
+
+    deleted_volume_ids = set(v.id for v in volumes)
+
+    def all_gone():
+        found_volume_ids = set(
+            v.id for v in cinder_volume_manager.list()
+        )
+        return len(deleted_volume_ids.intersection(found_volume_ids)) == 0
+    return poll_until(all_gone, repeat(1, 60))
+
+
```

Comment:
I'll disable these tests before merging.
They only make sense when run in an OpenStack environment where `osapi_max_limit = 5`.